atomic-exec/examples/fungible-token/src/lib.rs

```diff
@@ -90,7 +97,7 @@ pub struct ConstructorParams {
     /// Token ticker symbol.
     pub symbol: String,
     /// Initial balance table.
-    pub balances: HashMap<Address, TokenAmount>,
+    pub balances: HashMap<AddrString, TokenAmount>,
```

Comment:
What about adding a brief comment here to mention that we are using a HashMap because this is a test contract but that it would make more sense to use a HAMT if the number of balances expected is high?

Reply:
Can we use HAMT in a method parameter?

Reply:
Oh, right, the issue is that you need to provide the balances as a parameter. Using a HAMT then would require a slight re-architecture. Nevermind.

Reply:
The actor state maintains balances in a HAMT, though.